be/src/exec/olap_scanner.cpp

```diff
@@ -328,6 +328,11 @@ Status OlapScanner::get_batch(
                         slot->ptr = reinterpret_cast<char*>(v);
                     }
                 }
+
+                // the memory allocate by _read_row_cursor arena has been copied,
+                // so we should release these memory immediately
+                _read_row_cursor.reset_arena();
```

Comment:
You should do this in another place who will do aggregate finalize too. such as be/src/olap/merger.cpp which is used to do compaction

Reply:
OK